### ISSUE TITLE
feature request: `createBatchingNetworkInterface` should allow queries to unbatch

### ISSUE THREAD

**Issue by sodiumjoe** [apollographql/apollo-client#2056]
Use case: unblock fast calls from slow ones in a batched query

```js
const feedQuery = gql`{
  feed {
    stories {
      author { name }
      message
    }
  }
}`;

const commentsQuery = gql`{
  feed {
    stories {
      comments {
        author { name }
        message
      }
    }
  }
}`;

Promise.all([
  client.query({ query: feedQuery }).then(renderFeed);
  client.query({ query: commentsQuery).then(renderComments);
]);
```

In the above snippet, if the `commentsQuery` is slow, it will prevent the `renderFeed` function from being called because the queries are batched together (using `createBatchingNetworkInterface`).

I've worked around this by creating a custom networkInterface that separates known slow queries into their own batching interface:

```const networkInterface = {
	query: (request) => {
		if (_.includes(deferredQueries, request.operationName)) {
			return secondaryBatchingNetworkInterface.query(request);
		}
		return primaryBatchingNetworkInterface.query(request);
	},
};
```

It would be nice if I `batchingNetworkInterface` had native support for this that could handle this per call, rather than per query. I think a special variable like `batchingKey` would work. `batchingKey` could also be entirely opt-in, so this would not interfere whatsoever with current usage of `batchingNetworkInterface`.

**Comment by joelgriffith** [comment]
I'm curious if `apollo-client@2.x.x` will have mechanisms in which to implement stuff like this by consumers? I'd be interested in testing out 2.0.0 and writing a post about it since this will effectively solve some perf issues others' until things like deferred queries are a thing, or not :)

**Comment by sodiumjoe** [comment]
This still seems relevant. If a core team member has reviewed and decided not to do this, I'm ok with that, but i don't even know if anyone's looked at this.

**Comment by sodiumjoe** [comment]
> If you belive this issue is still a problem or should be reopened, please reopen it!

how do i do this
